app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,608 @@
+                if (currentSelected[itemKey]) {
+                    delete multiselectSelectedItems[itemKey];
+                } else {
+                    multiselectSelectedItems[itemKey] = item as Channel;
+                }
+
+                setMultiselectSelected(multiselectSelectedItems);
+                return;
+            }
+            default: {
+                if (currentSelected[itemKey]) {
+                    delete multiselectSelectedItems[itemKey];
+                } else {
+                    multiselectSelectedItems[itemKey] = item as DialogOption;
+                }
+                setMultiselectSelected(multiselectSelectedItems);
+            }
+        }
+    }, [integrationData, multiselectSelected, isMultiselect, dataSource, handleSelect]);
+
+    const handleRemoveOption = (item: UserProfile | Channel | DialogOption) => {
+        const currentSelected: Dictionary<UserProfile> | Dictionary<DialogOption> | Dictionary<Channel> = multiselectSelected;
+        const itemKey = extractItemKey(dataSource, item);
+        const multiselectSelectedItems = {...currentSelected};
+        delete multiselectSelectedItems[itemKey];
+        setMultiselectSelected(multiselectSelectedItems);
+    };
+
+    const getChannels = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {channels: channelData} = await fetchChannels(serverUrl, currentTeamId, page.current);
+
+            setLoading(false);
+
+            if (channelData && channelData.length > 0) {
+                setIntegrationData(channelData);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const getProfiles = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {users: profiles} = await fetchProfiles(serverUrl, page.current);
+
+            setLoading(false);
+
+            if (profiles && profiles.length > 0) {
+                setIntegrationData(profiles);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const loadMore = async () => {
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            await getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            await getChannels();
+        }
+
+        // dynamic options are not paged so are not reloaded on scroll
+    };
+
+    const searchDynamicOptions = useCallback((searchTerm = '') => {
+        if (options) {
+            setIntegrationData(options);
+        }
+
+        if (!getDynamicOptions) {
+            return;
+        }
+
+        getDynamicOptions(searchTerm.toLowerCase()).then((results: DialogOption[]) => {
```

Comment:
Any particular reason to use then instead of async / await?

Reply:
nope, I will async/await it.